### ISSUE TITLE
Firmware dump

### ISSUE THREAD

**Issue by Ingramz** [Ingramz/ecl110#3]
The main microcontroller used is Renesas R5F21258SN.

The programming and debug interface is supposedly done via 6 pins: VCC, GND, RX, TX, RST, MODE

Most of it is plain UART, however it looks like MODE and RST has to be used in a specific way to trigger the programming features.

[ToorCon 13 badge provides some instructions.](https://greatscottgadgets.com/tc13badge/). It should be possible to use a simple FTDI FT232 3.3V adapter to interface with the chip.

A quick glance at the datasheet shows that it is possible to "protect" chip from being read via 7 byte (56 bit) key. Obvious keys that should be tried are `00:00:00:00:00:00:00` and `FF:FF:FF:FF:FF:FF:FF`. If a different key is used, then an effort should be made to attempt to figure out the key, however as it is 56 bits, brute force will not be practical. A way to poke memory via modbus should be investigated in this case to extract or at worst overwrite the key.

If firmware binaries can be obtained from working units, this will enable upgrading early models to a newer firmware and further study its functions via inspecting the assembled binary.

**Comment by Ingramz** [comment]
Success!

![image](https://user-images.githubusercontent.com/1017132/45922161-42050b00-becd-11e8-9464-75b69f4e2e72.png)

**Comment by Ingramz** [comment]
# Help wanted!

I'm looking for someone who could provide me a dump of software version 1.08. I can provide the necessary tools for the job and know-how.

**Comment by mattiaslundin** [comment]
If you are still interested i can help out with 1.08

**Comment by Ingramz** [comment]
@mattiaslundin yes, definitely. Do you happen to own an FTDI serial adapter to perform he task outlined on the [firmware page](https://github.com/Ingramz/ecl110/blob/master/firmware.md)?

**Comment by mattiaslundin** [comment]
I’ve got an rs232 and 485 adapter but not sure if they’ll work. Will give it a try at least.

**Comment by Ingramz** [comment]
Plug them to your computer and see what the USB device ID-s are.

**Comment by Ingramz** [comment]
@mattiaslundin would you be still up for helping to dump the firmware?

**Comment by Kashelz** [comment]
> # Help wanted!
> 
> I'm looking for someone who could provide me a dump of software version 1.08. I can provide the necessary tools for the job and know-how.

@Ingramz  Hi. Do you know pass to unlock chip and read firmware for backup? I have 2 ECL110 devices with 116 setup v1.05 and v1.06. Want to convert them to 130, i have access with Flash Development Toolkit.

**Comment by Ingramz** [comment]
The two devices I've helped dump the firmware from so far have been both
unlocked - I think the default key was all FF.

I don't really know anything about the FDT, so I can't guarantee that the
raw dump on this repository is directly compatible with it. Pretty sure it
needs some modifications/cutting first, but I should be able to help out
once I know more about how FDT dumps look like.

My personal unit has been successfully upgraded from 1.06 to 1.08 and
working without any issues, so I can vouch that the procedure can be done
in principle.

Let me know if the key isn't working for you or you're having any other
issues.


E, 9. märts 2026, 22:59 Kashelz ***@***.***> kirjutas:

**Comment by Kashelz** [comment]
@Ingramz Thanks. Both my devices are locked then and non unlocked with all FF or 00. Chip itself can consist only 64kb of data, but your dumb is huge copared. I tried to make dump under virtual box linux your way, but without luck, i have USB TTL cable.

Is there a way to chat directly somehow, im interested to upgrade as i need to have 130 version and best with v1.08. I may have acces to another ECL with 130, but v1.05 or even older.

**Comment by Ingramz** [comment]
The dump is big because it contains the entire addressable memory space and some extra (0-FFFFF).

Easiest way to get hold of me is probably via [Discord](https://discordapp.com/users/168034266741997568) if that works for you.
